x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -335,15 +336,29 @@ export class FleetPlugin
     });
     licenseService.start(this.licensing$);
 
-    const fleetServerSetup = startFleetServerSetup();
-
     this.telemetryEventsSender.start(plugins.telemetry, core);
 
+    const logger = appContextService.getLogger();
+
+    const fleetSetupPromise = startFleetServerSetup()
+      .then(() =>
+        setupFleet(
+          new SavedObjectsClient(core.savedObjects.createInternalRepository()),
+          core.elasticsearch.client.asInternalUser
+        )
+      )
+      .then(() => {
+        logger.info('Fleet setup completed');
+      })
+      .catch((error) => {
```

Comment:
`catch`ing errors here seems to squash the `UnhandledPromiseRejection` errors that were breaking various tests in CI. 🤞